internal/hncconfig/reconciler.go

```diff
@@ -155,9 +156,12 @@ func (r *Reconciler) reconcileTypes(inst *api.HNCConfiguration) error {
 	// Overwrite the type set each time. Initialize them with the enforced types.
 	r.activeGVKMode = gr2gvkMode{}
 	r.activeGR = gvk2gr{}
-	if err := r.ensureEnforcedTypes(inst, allRes); err != nil {
-		// Early exit if any enforced types are not found for some reason to retry.
-		return err
+
+	if !config.EnforcedTypesDisabled {
+		if err := r.ensureEnforcedTypes(inst, allRes); err != nil {
```

Comment:
This should be an early return from inside `ensureEnforcedTypes` IMO. It's better not to clutter up the callers of the function if it can reasonable be handled _inside_ the function.